conf/machine/include/rpi-base.inc

```diff
@@ -76,7 +76,9 @@ MACHINE_FEATURES += "apm usbhost keyboard vfat ext2 screen touchscreen alsa blue
 # Raspberry Pi has no hardware clock
 MACHINE_FEATURES_BACKFILL_CONSIDERED = "rtc"
 
-MACHINE_ESSENTIAL_EXTRA_RRECOMMENDS += " kernel-modules udev-rules-rpi"
+MACHINE_EXTRA_RRECOMMENDS += "kernel-modules udev-rules-rpi"
+MACHINE_ESSENTIAL_EXTRA_RRECOMMENDS += "${@oe.utils.conditional("ENABLE_I2C", "1", "kernel-module-i2c-dev kernel-module-i2c-bcm2708", "", d)}"
```

Comment:
Small nitpick, I prefer to use ' inside " so that it doesn't confuse most text editors, but both versions work.

Thanks for follow-up.